velox/exec/Window.cpp

```diff
@@ -41,16 +41,78 @@ void initKeyInfo(
   }
 }
 
-void checkDefaultWindowFrame(const core::WindowNode::Function& windowFunction) {
-  VELOX_CHECK_EQ(
-      windowFunction.frame.type, core::WindowNode::WindowType::kRange);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.startType,
-      core::WindowNode::BoundType::kUnboundedPreceding);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.endType, core::WindowNode::BoundType::kCurrentRow);
-  VELOX_CHECK_EQ(windowFunction.frame.startValue, nullptr);
-  VELOX_CHECK_EQ(windowFunction.frame.endValue, nullptr);
+vector_size_t findFrameEndPointKPreceding(
+    core::WindowNode::WindowType type,
+    vector_size_t functionNumber,
+    vector_size_t partitionStartRow,
+    vector_size_t /*partitionEndRow*/,
+    vector_size_t currentRow,
+    std::optional<vector_size_t> offsetValue = 0) {
+  if (!offsetValue || type != core::WindowNode::WindowType::kRows) {
```

Comment:
This shouldn't be a per-row check. Move it to initialization time.

Reply:
Thanks for the suggestion, removed these redundant checks.